src/utils/windowUtils.ts

```diff
@@ -41,6 +41,35 @@ export module Utils {
       }
     }
 
+    /**
+     * Sends a warning to the console. The warning includes the version number of the deprecation,
+     * the name of the function which was deprecated and an optional message.
+     *
+     * Tu be used in the first line of a deprecated method.
+     *
+     * @param {string} version The version when the tagged method became obsolete
+     * @param {string?} message Optional message to be shown with the warning
+     */
+    export function deprecated(version: string, message = "") {
+
+      var callingMethod = "";
+
+      try {
+        // Getting the name of the calling method through the stack trace.
+        callingMethod = (<any> new Error).stack
+          .split("\n")
+          .filter((step: string) => step.match(/http/))[1] // Just method names
```

Comment:
There is a reference to the JS file, which always come from a http URL. This way we avoid the mocka injections and the extra lines (for Example Chrome also says Error on a line by itself).

Now thinking of it, I believe the deprecated method is so hacky, maybe I don't want to be so smart about it. Also using reflection is always a bad thing